docs/rate_limiter.md

```diff
@@ -0,0 +1,107 @@
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Rate Limiter
+
+Rate limiter manages the rate at which requests are scheduled on
+model instances by Triton. The rate limiter operates across all
+models loaded in Triton to allow *cross-model prioritization*.
+
+In absence of rate limiting (--rate-limit=off), Triton schedules
+execution of request (or set of requests when using batching) as
+soon as a model instance is available. This behavior is typically
+best suited for performance. However, there can be cases 
+where running all the models simultaneously places excessive load 
+on the server. For instance, model execution on some 
+frameworks dynamically allocate memory. Running all such models
+simultaneously may lead to system going out-of-memory.
+
+Rate limiter allows to post-pone the inference execution on some
+model instances such that not all of them runs simultaneously. 
+The model priorities are used to decide which model instance
+to schedule next. 
+
+## Using Rate Limiter
+
+To enable rate limiting users must set `--rate-limit` server
+cli option. For more information, consult usage of the option
+emitted by `tritonserver --help`.
+
+The rate limiter is controlled by rate limiter configuration
+described in [rate limiter config](model_configuration.md#rate-limiter-config).
+The rate limiter configuration includes 
+[resources](model_configuration.md#resources) and
+[priority](model_configuration.md#priority) for the instances
+defined by the instance group.
+
+### Resources
+
+Resources are just names with a corresponding count. By default
```

Comment:
"Resources are identified by a unique name and a count indicating the number of copies of the resource. By default, model instances use no...  the model instance indicates that is requires that many..."

When you are talking about the number of resources available I would use "resource copies" and when talking about model instances I would say "resource requirements". "count", "instance", "copy" are used here in multiple ways and that makes it confusing.

Reply:
Also might want to always say "model instance" to be extra clear